developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,311 @@
+
+### Signed transaction
+A signed transaction is consisted of: a raw transaction and an authenticator. An authenticator contains a client's public key and the signature of the raw transaction.
+
+After BCS serialization and hex-coding, signed transactions are ready for submission.
+
+## Detailed Steps
+
+1. Create a RawTransaction
+2. Create a signing message
+3. Sign the signing message with account private key
+4. Create a SignedTransaction
+5. Serialize SignedTransaction into bytes with BCS, and then hex-encode the bytes into a string
+
+## Step 1) Create a RawTransaction
+
+```typescript
+interface TransactionPayload {
+  type: string, // Payload types, e.g. "script_function_payload"
+  function: string, // Fully qualified function name, e.g. "0x1::TestCoin::transfer"
+  type_arguments: string[], // Type arguments for generic functions
+  arguments: string[] // Arguments for the function to run
+}
+
+interface RawTransaction {
+  // hex-coded account address of the sender, e.g. "0x00000000000000000000000000000001"
+  sender: string,
+  // Sequence number of this transaction. This must match the sequence number
+  // stored in the sender's account at the time the transaction executes.
+  sequence_number: number,
+
+  payload: TransactionPayload,
+
+  // Maximal total gas to spend for this transaction.
+  max_gas_amount: number,
+
+  // Price to be paid per gas unit.
+  gas_unit_price: number,
+
+  // Expiration timestamp for this transaction, represented
+  // as seconds from the Unix Epoch. If the current blockchain timestamp
+  // is greater than or equal to this time, then the transaction has
+  // expired and will be discarded. This can be set to a large value far
+  // in the future to indicate that a transaction does not expire.
+  expiration_timestamp_secs: number,
+
+  // Chain ID of the Aptos network this transaction is intended for.
+  // MAINNET = 1, TESTNET = 2, DEVNET = 3, TESTING = 4, PREMAINNET = 5
+  chain_id: number,
+}
+
+function createRawTransaction(): RawTransaction {
+  const payload: TransactionPayload = {
+    type: "script_function_payload",
+    function: "0x1::TestCoin::transfer",
+    type_arguments: [],
+    arguments: [
+      `0x00000000000000000000000000000002`, // receipient of the transfer
+      "1", // amount to transfer
+    ]
+  }
+
+  return {
+    "sender": "0x00000000000000000000000000000001",
+    "sequence_number": 1,
+    "max_gas_amount": 2000,
+    "gas_unit_price": 1,
+    // Unix timestamp, in seconds + 10 minutes
+    "expiration_timestamp_secs": Math.floor(Date.now() / 1000) + 600,
+    "payload": payload,
+    "chain_id": 3
+  };
+}
+
+```
+
+## Step 2) Create a signing message
+
+1. [SHA3_256][sha3] hash bytes of string "APTOS::RawTransaction"
+2. Bytes of BCS serialized RawTransaction
+3. Concat the prefix and BCS bytes
+
+```typescript
+
+class BCS {
```

Comment:
I don't think we should be adding this much code here, I think showing how to generate a salt and pointing to a library for bcs should be sufficient.

Reply:
👍